### DISCUSSION TITLE
Masking out values above N as no-data

### DISCUSSION THREAD

**Discussion by neelduttahere** [developmentseed/titiler#745] [Q&A] [answered]
Hi, here is the problem statement I am concerned with:

Lets say I have a single band tiff file with values inside the range of 0-8. Is there any way that I can remove (or make it appear transparent) all values above 6? Or force titiler to consider all values above 6 to be no-data? So that we can only view the pixels within 0-5 of the tiff file, and the color map acting up accordingly, and anything 6 or above will appear transparent?

I was thinking of using custom json colormaps, but then I would have to visualize my output in discrete buckets, instead of a continuous range.

I would like to know anyone's thoughts on this.

**Comment by vincentsarago** [comment]
@neelduttahere basically you want to tell that all data >6 are like nodata?

2 Solutions:

- `interval colormap` https://cogeotiff.github.io/rio-tiler/colormap/#intervals-colormaps but as mentioned you'll have to write discrete bucket 
- custom algorithm https://developmentseed.org/titiler/advanced/Algorithms/

**Comment by neelduttahere** [comment]
> @neelduttahere basically you want to tell that all data >6 are like nodata?
> 
> 2 Solutions:
> 
> * `interval colormap` https://cogeotiff.github.io/rio-tiler/colormap/#intervals-colormaps but as mentioned you'll have to write discrete bucket
> * custom algorithm https://developmentseed.org/titiler/advanced/Algorithms/

Yes, all > 6 should be no data. Interval colormap is not might desired approach but checking on the second one.

**Reply by vincentsarago** [reply, marked as answer]
Yeah so the custom `Algorithm` route will be what you are looking for!
